### PR TITLE
Internationalization: Translate VariableInput and VariableOptions components

### LINKED ISSUE #58739: Translate VariableOptions
In a `dashboard` with a `template variable`, we can see the variable selector to choose a value. In `VariableOptions` we can translate the `Selected` value and its `tooltip`.

VariableOption

### LINKED ISSUE #58740: Translate VariableInput
In a dashboard with a template variable, we can see the variable selector to choose a value. In `VariableInput` we can translate the placeholder and in `VariableLink` the `'All'`.

VariableInput

### PR DESCRIPTION
**Pull request by eledobleefe**
**What this PR does / why we need it:**
In the process of getting ready for the translation of the portal, we need to extract strings into translation tags. One of our first steps is to translate the Dashboard in view-only mode. This ticket does part of that by translating the texts related to the template variable value selector.

**Which issue(s) this PR fixes:**

Fixes #58740 and #58739 

VariableOption

VariableInput

### CODE REVIEW

File: public/app/features/variables/pickers/shared/VariableInput.tsx [hunk 1]
```diff
@@ -37,7 +39,7 @@ export class VariableInput extends PureComponent<Props> {
         value={value ?? ''}
         onChange={this.onChange}
         onKeyDown={this.onKeyDown}
-        placeholder="Enter variable value"
+        placeholder={t('variable.picker.input', 'Enter variable value')}
```
Comment (JoaoSilvaGrafana) [inline public/app/features/variables/pickers/shared/VariableInput.tsx:42]: Maybe `variable.picker.input-placeholder` to keep the context?

**Review by JoaoSilvaGrafana — COMMENTED** [review]
How about the "All" string, do we translate that here as well?

File: public/app/features/variables/pickers/shared/VariableInput.tsx (hunk 1 above)
Comment (eledobleefe) [reply public/app/features/variables/pickers/shared/VariableInput.tsx:42]: Yes @JoaoSilvaGrafana. I've just done it. It is changed in `VariableOptions` and `VariableList`. I run it locally and it works. Are they ok?

File: public/app/features/variables/pickers/shared/VariableLink.tsx [hunk 2]
```diff
@@ -65,7 +66,7 @@ interface VariableLinkTextProps {
 
 const VariableLinkText: FC<VariableLinkTextProps> = ({ text }) => {
   const styles = useStyles2(getStyles);
-  return <span className={styles.textAndTags}>{text}</span>;
+  return <span className={styles.textAndTags}>{text === 'All' ? t('variable.picker.link-all', 'All') : text}</span>;
```
[the remark this answers is not in the document]
Comment (eledobleefe) [reply public/app/features/variables/pickers/shared/VariableLink.tsx]: Good to know @JoaoSilvaGrafana! I changed the code to use the constants. Thanks!

**Review by JoaoSilvaGrafana — APPROVED** [review]
🎉

**Pull request merged**